### ISSUE TITLE
Explanation on comparison filter for 2D variables not so clear

### ISSUE THREAD

**Issue by StevenCompernolle** [stcorp/harp#297]
Currently the following is written (http://stcorp.github.io/harp/doc/html/operations.html)

_Filter a dimension for all variables in the product such that items for which the value of the provided variable does not match the expression get excluded. The variable should be one dimensional or two dimensional (with the first dimension being time). The dimension that gets filtered is the last dimension of the referenced variable._

It was not so clear to me what would happen for 2D `{time, other}` variables from this explanation, so I did a few tests: on a Cloudnet {time,vertical} cloud_type variable (hence integer) and on a {time,vertical} prior profile.

What the operation seems to do is:

- in the second dimension: keep only the values that obey the condition. This also means that along the first (time) dimension, the position of the second dimension is no longer consistent (e.g., the indices along dimension vertical of the first and of the second profile do no longer correspond to the same vertical position). The number of elements excluded can differ along the time dimension, therefore trailing positions along the second dimension are 0-filled (for an integer variable) or nan-filled (for a float variable).
- in the time dimension: positions along the time dimension, where the condition along the second dimension is never fulfilled, are removed.

So in a sense this works as an `np.any` filter: Pixels where the condition on the second dimension is never fulfilled, are removed.

Can you confirm that it works as described above? And if so, perhaps update the description?

Would it also work for {time,independent} string variables? 
I could then e.g., have a `location_name {time, independent}` variable, where the same pixel can be shared by different overpasses, and using the operation `location_name=='bremen'` would select only these wich have 'bremen' in location_name (irrespective if there are also other names in this position)?

**Comment by svniemeijer** [comment]
This aspect is actually described at the end of http://stcorp.github.io/harp/doc/html/conventions/dimensions.html
It is a generic HARP convention that should hold for all HARP operations.

Filtering on the `independent` dimension is not allowed. Wherever you currently do not get an error for this, this should be considered a bug (as was the case for #295).

**Comment by svniemeijer** [comment]
Regarding your `location_name {time, independent}` question, be aware of the difference between the 'internal' HARP product representation and the one as exported to netcdf3: http://stcorp.github.io/harp/doc/html/conventions/netcdf3.html
If with `independent` you mean the additional dimension that gets added for strings in the netcdf3 export, then you should be able to do `location_name=="bremen"` to filter the `time` dimension for entries where you have that location, because internally the variable is just a `location_name {time}` variable.

**Comment by StevenCompernolle** [comment]
No, I really meant `location_name {time, independent}` as represented internally. To be able to store several names in location_name for the same pixel (since overpasses can overlap). But I understand this is not possible, because it involves an 'independent' dimension?

**Comment by svniemeijer** [comment]
Indeed. That won't work. You would either have to replicate the pixel (with same timestamp, geolocation, value, etc. except for `location_name {time}`) or, if your set of locations is limited, you could use a bitmask field (location1 = 1, location2 = 2, location3 = 4, location4 = 8, etc.), so you can use `location=&4' to filter all measurements that match with location3.
